network/framework/src/peer/mod.rs

```diff
@@ -81,13 +81,33 @@ pub enum PeerRequest {
 pub enum DisconnectReason {
     Requested,
     ConnectionLost,
+    FailedHealthCheckPing(PingDisconnectContext),
+    FailedPeerMonitoringPing(PingDisconnectContext),
+}
+
+/// Ping Disconnect Reason
```

Comment:
nit: `Ping Disconnect Reason` -> `Ping Disconnect Context`? 😄

Reply:
Haha will change the comment... Context... for the reason... :D